src/commands/map.test.ts

```diff
@@ -0,0 +1,450 @@
+            },
+          })
+        ).rejects.toThrow(``);
+      });
+    });
+
+    describe('checking profile id argument', () => {
+      it('throws when profile id is not provided', async () => {
+        jest.mocked(exists).mockResolvedValueOnce(true);
+        jest
+          .mocked(readFile)
+          .mockResolvedValueOnce(JSON.stringify(providerJson));
+        await expect(
+          instance.execute({
+            logger,
+            userError,
+            flags: {},
+            args: { providerName },
+          })
+        ).rejects.toThrow(
+          'Missing profile id. Please provide it as first argument.'
+        );
+      });
+
+      it('throws when profile id is invalid', async () => {
+        jest.mocked(exists).mockResolvedValueOnce(true);
+        jest
+          .mocked(readFile)
+          .mockResolvedValueOnce(JSON.stringify(providerJson));
+        await expect(
+          instance.execute({
+            logger,
+            userError,
+            flags: {},
+            args: { providerName, profileId: '!_0%L' },
+          })
+        ).rejects.toThrow(
+          'Invalid profile id: "!_0%L" is not a valid lowercase identifier'
+        );
+      });
+
+      it('throws when profile file does not exist', async () => {
+        jest
+          .mocked(exists)
+          .mockResolvedValueOnce(true)
+          .mockResolvedValueOnce(false);
+        jest
+          .mocked(readFile)
+          .mockResolvedValueOnce(JSON.stringify(providerJson));
+        await expect(
+          instance.execute({
+            logger,
+            userError,
+            flags: {},
+            args: { providerName, profileId: '!_0%L' },
+          })
+        ).rejects.toThrow(
+          'Invalid profile id: "!_0%L" is not a valid lowercase identifier'
+        );
+      });
+
+      it('throws when reading of file fails', async () => {
+        jest
+          .mocked(exists)
+          .mockResolvedValueOnce(true)
+          .mockResolvedValueOnce(true);
+        jest
+          .mocked(readFile)
+          .mockRejectedValueOnce(new Error('File read error'));
+        await expect(
+          instance.execute({
+            logger,
+            userError,
+            flags: {},
+            args: { providerName, profileId: profileName },
+          })
+        ).rejects.toThrow('File read error');
+      });
+
+      it('throws when profile source is not valid Comlink', async () => {
+        jest
+          .mocked(exists)
+          .mockResolvedValueOnce(true)
+          .mockResolvedValueOnce(true);
+        jest
+          .mocked(readFile)
+          .mockResolvedValueOnce(JSON.stringify(providerJson))
+          .mockResolvedValueOnce('something');
+        await expect(
+          instance.execute({
+            logger,
+            userError,
+            flags: {},
+            args: { providerName, profileId: profileName },
+          })
+        ).rejects.toThrow();
```

Comment:
question: User will not get meaningful error message?

Reply:
He will get meaningfull message:)